x/mongo/driver/connstring/connstring.go

```diff
@@ -21,7 +21,24 @@ import (
 	"go.mongodb.org/mongo-driver/x/mongo/driver/wiremessage"
 )
 
-// Parse parses the provided uri and returns a URI object.
+// ParseAndValidate parses the provided uri and returns a URI object.
+// It check that all values are valid.
+func ParseAndValidate(s string) (ConnString, error) {
+	p := parser{dnsResolver: dns.DefaultResolver}
+	err := p.parse(s)
+	if err != nil {
+		return p.ConnString, internal.WrapErrorf(err, "error parsing uri")
+	}
+	err = p.ConnString.Validate()
+	if err != nil {
+		return p.ConnString, internal.WrapErrorf(err, "error parsing uri")
+	}
+	return p.ConnString, nil
+}
+
+// Parse parses the provided uri and returns a URI object
+// but does not check that all values are valid. Use `ConnString.ParseAndValidate()`
```

Comment:
Was this supposed to say "Use `ConnString.Validate` to run validation checks separately"?

Reply:
yep